superset/models/helpers.py

```diff
@@ -1378,6 +1381,8 @@ def values_for_column(
             sql = self.mutate_query_from_config(sql)
 
             df = pd.read_sql_query(sql=sql, con=engine)
+            # replace NaN with None to ensure it can be serialized to JSON
+            df = df.replace({np.nan: None})
```

Comment:
I'm curious, do we ever _not_ want to do this?

Reply:
What I'm somewhat perplexed about is why historically this hasn't been an issue? I'm also not entirely sure when  the `values_for_column` function is invoked.

Note it seems like Trino et al. support [NaNs](https://trino.io/docs/current/functions/math.html), though interesting in SQL Lab if I run `SELECT nan()` the result set renders `NULL`.

Reply:
@john-bodley this only happens in numeric columns with `NULL` values. The function is called when populating values for a filter dropdown. My assumption is that we haven't noticed this because numeric columns are not commonly used for filters; most of them are strings.

The biggest issue here, which my PR does not address, is that our backend JSON encoder is happily encoding the `NaN`s like this:

```json
[1, NaN, 3]
```

This is not valid JSON as far as I know (the spec is unclear on `null` and `NaN`, which are Javascript objects but technically not valid JSON).

Reply:
Thanks @betodealmeida for clarifying. Who doesn't love the whole encoding/decoding of these more atypical types?

Reply:
TBH I had forgotten JS actually has a `NaN` 😁